contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,751 @@
+    )
+        external
+        onlyAdminACL(this.addDependencyTypeAdditionalCDN.selector)
+        onlyNonEmptyString(_additionalCDN)
+    {
+        uint24 additionalCDNCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalCDNCount;
+        dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+            additionalCDNCount
+        ] = _additionalCDN;
+        dependencyTypeInfo[_dependencyTypeId].additionalCDNCount =
+            additionalCDNCount +
+            1;
+
+        emit DependencyTypeAdditionalCDNUpdated(
+            _dependencyTypeId,
+            _additionalCDN,
+            additionalCDNCount
+        );
+    }
+
+    /**
+     * @notice Removes additional CDN for depenency `_dependencyId` at index `_index`.
+     * Removal is done by swapping the element to be removed with the last element in the array, then deleting this last element.
+     * Assets with indices higher than `_index` can have their indices adjusted as a result of this operation.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _index Additional CDN index
+     */
+    function removeDependencyTypeAdditionalCDNAtIndex(
+        bytes32 _dependencyTypeId,
+        uint256 _index
+    )
+        external
+        onlyAdminACL(this.removeDependencyTypeAdditionalCDNAtIndex.selector)
+        onlyExistingDependencyType(_dependencyTypeId)
+    {
+        uint24 additionalCDNCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalCDNCount;
+        require(_index < additionalCDNCount, "Asset index out of range");
+
+        uint24 lastElementIndex = additionalCDNCount - 1;
+
+        dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+                _index
+            ] = dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+            lastElementIndex
+        ];
+        delete dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+            lastElementIndex
+        ];
+
+        dependencyTypeInfo[_dependencyTypeId]
+            .additionalCDNCount = lastElementIndex;
+
+        emit DependencyTypeAdditionalCDNRemoved(_dependencyTypeId, _index);
+    }
+
+    /**
+     * @notice Updates additional CDN for dependency type `_dependencyTypeId` at `_index`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _index Additional CDN index.
+     * @param _additionalCDN New CDN URL.
+     */
+    function updateDependencyTypeAdditionalCDNAtIndex(
+        bytes32 _dependencyTypeId,
+        uint256 _index,
+        string memory _additionalCDN
+    )
+        external
+        onlyAdminACL(this.updateDependencyTypeAdditionalCDNAtIndex.selector)
+    {
+        uint24 additionalCDNCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalCDNCount;
+        require(_index < additionalCDNCount, "Asset index out of range");
+
+        dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+            _index
+        ] = _additionalCDN;
+
```

Comment:
I think need to increment `dependencyTypeInfo[_dependencyTypeId].additionalCDNCount`?

Reply:
nvm - this is an update, not add - resolving